chat-plugins/badges.js

```diff
@@ -53,43 +54,72 @@ exports.commands = {
-			if (!parts[1]) return this.errorReply("Invalid command. Valid commands are `/badges list`, `/badges info, badgeName`, `/badges set, user, badgeName` and `/badges take, user, badgeName`.");
-			selectedBadge = parts[1].trim();
+			if (!target) return this.errorReply("Correct command: `/badges info badgeName`.");
+			selectedBadge = target.trim();
 			if (!Db('badgeData').has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list");
 			let badgeData = Db('badgeData').get(selectedBadge);
 			this.sendReplyBox(badgeImg(badgeData[1], selectedBadge) + selectedBadge + ': ' + badgeData[0]);
-			break;
-		case 'take':
+		},
+		take: function (target, room, user) {
+			let parts = target.split(',');
+			let userid;
+			let selectedBadge;
+			let userBadges;
 			if (!this.can('lock')) return false;
-			if (parts.length !== 3) return this.errorReply("Correct command: `/badges take, user, badgeName`");
-			userid = toId(parts[1].trim());
+			if (parts.length !== 2) return this.errorReply("Correct command: `/badges take user, badgeName`");
+			userid = toId(parts[0].trim());
 			if (!Db('userBadges').has(userid)) return this.errorReply("This user doesn't have any badges.");
 			userBadges = Db('userBadges').get(userid);
-			selectedBadge = parts[2].trim();
+			selectedBadge = parts[1].trim();
 			userBadges = userBadges.filter(b => b !== selectedBadge);
 			Db('userBadges').set(toId(userid), userBadges);
+
 			this.logModCommand(user.name + " took the badge '" + selectedBadge + "' badge from " + userid + ".");
 			this.sendReply("The '" + selectedBadge + "' badge was taken from '" + userid + "'.");
-			break;
-		case 'delete':
+		},
+		delete: function (target, room, user) {
+			let selectedBadge;
 			if (!this.can('ban')) return false;
-			if (parts.length !== 2) return this.errorReply("Correct command: `/badges deleteall, badgeName`");
-			selectedBadge = parts[1].trim();
+			if (!target) return this.errorReply("Correct command: `/badges deleteall badgeName`");
+			selectedBadge = target.trim();
 			if (!Db('badgeData').has(selectedBadge)) return this.errorReply("This badge does not exist, please check /badges list");
 			Db('badgeData').delete(selectedBadge);
 			let badgeUserObject = Db('userBadges').object();
 			let users = Object.keys(badgeUserObject);
 			users.forEach(u => Db('userBadges').set(u, (badgeUserObject[u].filter(b => b !== selectedBadge))));
+
 			this.sendReply("The badge with the name '" + selectedBadge + "' deleted.");
 			this.logModCommand(user.name + " removed the badge '" + selectedBadge + ".");
-			break;
-		default:
-			return this.errorReply("Invalid command. Valid commands are `/badges list`, `/badges info, badgeName`, `/badges set, user, badgeName` and `/badges take, user, badgeName`" +
-			 "`/badges create, name, description, img`.");
-		}
+		},
+		transfer: function (target, room, user) {
+			let parts = target.split(',');
+			if (!this.can('ban')) return false;
```

Comment:
Why?

This was the entire reason I said to check for user object; so users can do it themselves...

Then again, if someone merges with another user, that could be problematic. That could still be problematic this way, too, though... Hmm...